ai-decision-services/ai-ad-dpp-engine/prepare-config-data-and-app/prepare-config-data-and-app.md

```diff
@@ -0,0 +1,217 @@
+
+## inputSources
+
+Dataset(s) input source(s), with **inputSources** as key and an array of dataset(s) information as value.
+
+*   **dataframeName** - the variable name of this data frame will be used in the whole context.
+*   **type** - can either be **object-storage** or **oracle**.
+*   **isEventDriven** - (optional) This is an optional parameter whose default value is _false_. It takes a boolean value of _true_ is provided if the input source needs to be taken from a configured event. Note that currently, it is only supported for Object Storage emitted events.
+*   For object storage, you will require:
+    *  **namespace** \- the namespace of the target object storage bucket. 
+    *   **bucket** - the bucket name. 
+    *   **objectName** \- the full path and name of the object. This can also be a prefix with star (\*) when _isEventDriven_ is provided with _true_ value, where the object generating the event is considered and attempted to read if the name matches the provided expression. For example, updating object with name _anomalies/test.csv_ can generate an event and can be captured by using _anomalies/\*_ as the objectName, provided isEventDriven is set to true. Also, note that anything after \* is ignored in our current version and we will add more functionality in upcoming sprints.
+*   For oracle, you will require:
+    *   **adbId** \- the OCID of your ADW application. 
+    *   **tableName** \- name of the table. 
+    *   **connectionId** \- the format is **<DBName>\_high**, which can be found under tnsnames.ora of [Wallet](https://docs.oracle.com/en/cloud/paas/autonomous-database/adbsa/connect-download-wallet.html#GUID-B06202D2-0597-41AA-9481-3B174F75D4B1). 
+    *   **user** \- the owner of the table. In most cases it will be **admin**. 
+    *   **password** \- the password you setup when you download the Wallet. 
+
+## phaseInfo
+
+The flag indicating whether it is a training or inferencing phase, plus other required input parameters. Here we will include an object storage connector, the workflow will use it to store or read metadata, which is useful for data dependent steps (for example, one-hot encoding).
+
+*   **connector** - the connector to object storage. It will be composed of
+    *   **namespace** - the namespace of the target object storage bucket. Required if it is for object storage.
+    *   **bucket** - the bucket name. Required if it is for object storage.
+    *   **objectName** \- the full path and name of the object. Required if it is for object storage.
```

Comment:
These come from the Object Storage setup step. Please reorder this to split the object storage setup into 2 parts - one for driver, archive & data sets and the second for the config. The first part should come before the config file preparation and the second should come after

Reply:
In that case, maybe it makes more sense to change the lab sequence and let user prepare configurations at the end. Since all resources should be ready before users can work on a configuration. What do you think? @ganrad

Reply:
I think our current flow is mostly taking that into account, except for the step where you upload the driver and archive to Object Storage buckets. If we reorganize that a bit, I think it will be correct. Since we don't refer to Events and Function OCIDs or Data Flow OCIDs anywhere in our config, it should be okay to have those Labs come later. Train/test data will not be uploaded until we are ready to run the test, so that can remain in Lab 8(run-tests)